src/envoy/alts/alts_socket_factory.cc

```diff
@@ -29,6 +29,36 @@ namespace Envoy {
 namespace Server {
 namespace Configuration {
 
+using ::google::protobuf::RepeatedPtrField;
+
+// Returns true if the peer's service account is found in targets, otherwise
+// returns false and fills out err with an error message.
+static bool doValidate(const tsi_peer &peer,
+                       const RepeatedPtrField<std::string> &targets,
+                       std::string &err) {
+  for (size_t i = 0; i < peer.property_count; ++i) {
+    std::string name = std::string(peer.properties[i].name);
+    std::string value = std::string(peer.properties[i].value.data,
+                                    peer.properties[i].value.length);
+    if (name.compare(TSI_ALTS_SERVICE_ACCOUNT_PEER_PROPERTY) == 0) {
+      for (int j = 0; j < targets.size(); ++j) {
+        if (value.compare(targets[j]) == 0) {
+          return true;
+        }
+      }
+    }
+  }
+
+  err = "Couldn't find peer's service account in targets: ";
+  for (int i = 0; i < targets.size(); ++i) {
```

Comment:
For easy debugging, it is better to log both peers and targets.

Reply:
The peers will be logged in src/envoy/alts/tsi_transport_socket.cc just before the handshake_validator is called.